src/libs/ajax/User.ts

```diff
@@ -0,0 +1,398 @@
+              subject,
+              // BEWARE changing the following ids or values! If you change them then you must thoroughly test.
+              custom_fields: [
+                { id: 360012744452, value: type },
+                { id: 360007369412, value: description },
+                { id: 360012744292, value: name },
+                { id: 360012782111, value: email },
+                { id: 360018545031, value: emailAgreed },
+                { id: 360027463271, value: clinicalUser },
+              ],
+              comment: {
+                body: `${description}\n\n------------------\nSubmitted from: ${currUrl}`,
+                uploads: [`${attachmentToken}`],
+              },
+            },
+          })
+        )
+      );
+    },
+
+    uploadAttachment: async (file) => {
+      const res = await fetchOk(`https://support.terra.bio/api/v2/uploads?filename=${file.name}`, {
+        method: 'POST',
+        body: file,
+        headers: {
+          'Content-Type': 'application/binary',
+        },
+      });
+      return (await res.json()).upload;
+    },
+
+    firstTimestamp: (): RexFirstTimestampResponse => {
+      return getFirstTimeStamp(getTerraUser().token);
+    },
+
+    getNihStatus: async (): Promise<OrchestrationNihStatusResponse | undefined> => {
+      try {
+        const res = await fetchOrchestration('api/nih/status', _.merge(authOpts(), { signal }));
+        return res.json();
+      } catch (error: any) {
+        if (error.status === 404) {
+          return;
+        }
+        throw error;
+      }
+    },
+
+    linkNihAccount: async (token: string): Promise<OrchestrationNihStatusResponse> => {
+      const res = await fetchOrchestration(
+        'api/nih/callback',
+        _.mergeAll([authOpts(), jsonBody({ jwt: token }), { signal, method: 'POST' }])
+      );
+      return res.json();
+    },
+
+    unlinkNihAccount: async (): Promise<void> => {
+      await fetchOrchestration('api/nih/account', _.mergeAll([authOpts(), { signal, method: 'DELETE' }]));
+    },
+
+    getFenceStatus: async (provider): Promise<BondFenceStatusResponse | {}> => {
+      try {
+        const res = await fetchBond(`api/link/v1/${provider}`, _.merge(authOpts(), { signal }));
+        return res.json();
+      } catch (error: any) {
+        if (error.status === 404) {
+          return {};
+        }
+        throw error;
+      }
+    },
+
+    getFenceAuthUrl: async (providerKey: string, redirectUri: string): Promise<BondFenceUrlResponse> => {
+      const queryParams = {
+        scopes: ['openid', 'google_credentials', 'data', 'user'],
+        redirect_uri: redirectUri,
+        state: btoa(JSON.stringify({ providerKey })),
+      };
+      const res = await fetchBond(
+        `api/link/v1/${providerKey}/authorization-url?${qs.stringify(queryParams, { indices: false })}`,
+        _.merge(authOpts(), { signal })
+      );
+      return res.json();
+    },
+
+    linkFenceAccount: async (
```

Comment:
Are there types for the inputs to `linkFenceAccount` and `unlinkFenceAccount`?

Reply:
I had a hard time finding them at first, but when I did, I forgot to update them here (the provider being named provider is really awful).